### PR TITLE
Backport HIVE-19048: Initscript errors are ignored

### PR DESCRIPTION
**Pull request by turboFei**
### _Why are the changes needed?_

FYI: 
For hive-3.1.3, if failed to get connection, the return code of `beeline -f` is 0, the correct one should be non-zero.
And it works well for hive-4.0.0-alpha-2.
image

And the hive-beeline dependency version for kyuubi is 3.1.3.

This pr backports HIVE-19048: Initscript errors are ignored.

https://issues.apache.org/jira/browse/HIVE-19048
https://github.com/apache/hive/commit/006bf8a1a49aabf9930744c977fb0b9c91a5ef6c

### _How was this patch tested?_
- [x] Add some test cases that check the changes thoroughly including negative and positive cases if possible

- [ ] Add screenshots for manual tests if appropriate

- [x] [Run test](https://kyuubi.readthedocs.io/en/master/develop_tools/testing.html#running-tests) locally before make a pull request

### CODE REVIEW

File: kyuubi-hive-beeline/src/main/java/org/apache/hive/beeline/KyuubiBeeLine.java [hunk 1]
```diff
@@ -189,4 +199,53 @@ int initArgs(String[] args) {
...
+            exitField.set(exit);
+          } catch (Exception t) {
+            error(t.getMessage());
+            return ERRNO_OTHER;
+          }
+        }
+      }
+    }
+    return executionResult;
+  }
```
Comment (turboFei) [inline kyuubi-hive-beeline/src/main/java/org/apache/hive/beeline/KyuubiBeeLine.java:250]: The code in hive-3.1.3.
image

File: kyuubi-hive-beeline/src/main/java/org/apache/hive/beeline/KyuubiBeeLine.java [hunk 2]
```diff
@@ -36,6 +36,11 @@ public class KyuubiBeeLine extends BeeLine {
   protected KyuubiCommands commands = new KyuubiCommands(this);
   private Driver defaultDriver;
 
+  // copied from org.apache.hive.beeline.BeeLine
+  private static final int ERRNO_OK = 0;
+  private static final int ERRNO_ARGS = 1;
+  private static final int ERRNO_OTHER = 2;
```
Comment (turboFei) [inline kyuubi-hive-beeline/src/main/java/org/apache/hive/beeline/KyuubiBeeLine.java:42]: copied.

image

**Review by pan3793 — APPROVED** [review]

**Review by cxzl25 — APPROVED** [review]

**Comment by pan3793** [comment]
Thanks, merged to master/1.7

**Comment by pan3793** [comment]
Some conflict on 1.7, please make PR independently.

**Comment by pan3793** [comment]
backported to 1.7 via ec00001090f0c15ae5dcd7c6e11fa6e3d9aaee1e

**Pull request closed without merging**